src/services/blocks/BlocksService.ts

```diff
@@ -98,22 +101,24 @@ export class BlocksService extends AbstractService {
 			return isBlockCached;
 		}
 
-		const [deriveBlock, events, finalizedHead] = await Promise.all([
-			api.derive.chain.getBlock(hash),
-			this.fetchEvents(api, hash),
+		const [{ block }, validators, events, finalizedHead] = await Promise.all([
+			api.rpc.chain.getBlock(hash),
```

Comment:
Is there a reason we don't completely get of `api` and instead use `historicApi` everywhere?

Like at the top of all these services we could replace `const { api } = this;` with `const  api = this.api.at(has)`;

Reply:
So we still need to use the api for rpc methods, which is why we would keep the `this.api`. Whats good is `this.api` will cache the registries for each runtime it creates a historicApi for, so it gets pretty fast.